qa/tasks/cephfs/caps_helper.py

```diff
@@ -193,3 +193,37 @@ def conduct_neg_test_for_write_caps(self, sudo_write=False):
             cmdargs.pop(-1)
             log.info('absence of write perm was tested successfully: '
                      f'failed to be write data to file {path}.')
+
+    def _conduct_neg_test_for_root_squash_caps(self, _cmdargs, sudo_write=False):
+        possible_errmsgs = ('permission denied', 'operation not permitted')
+        cmdargs = ['sudo'] if sudo_write else ['']
+        cmdargs += _cmdargs
+
+        for mount, path, data in self.test_set:
+            log.info(f'test absence of {_cmdargs[0]} perm: expect failure {path}.')
+
+            # open the file and hold it. The MDS will issue CEPH_CAP_EXCL_*
+            # to mount
+            proc = mount.open_background(path)
+            mount.wait_for_visible(path)
+
+            cmdargs.append(path)
+            mount.negtestcmd(args=cmdargs, retval=1, errmsgs=possible_errmsgs)
+            cmdargs.pop(-1)
+
+            mount._kill_background(proc)
+
+            log.info(f'absence of {_cmdargs[0]} perm was tested successfully')
+
+    def conduct_neg_test_for_chown_caps(self, sudo_write=True):
+        # flip ownership to nobody. assumption: nobody's id is 65534
+        cmdargs = ['chown', '-h', '65534:65534']
+        self._conduct_neg_test_for_root_squash_caps(cmdargs, sudo_write)
+
+    def conduct_neg_test_for_truncate_caps(self, sudo_write=True):
+        cmdargs = ['truncate', '-s', '10GB']
+        self._conduct_neg_test_for_root_squash_caps(cmdargs, sudo_write)
+
+    def conduct_neg_test_for_open_caps(self, sudo_write=True):
+        cmdargs = ['cat']
```

Comment:
So we can't `cat` files as sudo with root_squash caps anymore?  Doesn't this contradict

https://github.com/ceph/ceph/blob/4ede08e33632b8dbcff0e64115d5ea5232b4a41c/qa/tasks/cephfs/test_admin.py#L1230-L1233

Reply:
The commit https://github.com/ceph/ceph/pull/48027/commits/0cf1180db682ff24af388c5211d96673248dd278 seems buggy, which should prevent it when trying to open the file by truncating it. Will fix it.